Draft: Align UA Stylesheet with MathML Core for maction and semantics.

eerii opened the pull request:
#### cded5d0e51a96a9161bed09c89bc9198e2df6018
<pre>
Align semantics and maction with MathML Core
<a href="https://bugs.webkit.org/show_bug.cgi?id=262696">https://bugs.webkit.org/show_bug.cgi?id=262696</a>

Reviewed by NOBODY (OOPS!).

The MathML Core spec says that semantics
(<a href="https://w3c.github.io/mathml-core/#semantics-and-presentation)">https://w3c.github.io/mathml-core/#semantics-and-presentation)</a> and
maction (<a href="https://w3c.github.io/mathml-core/#dfn-maction)">https://w3c.github.io/mathml-core/#dfn-maction)</a> have the same
layout algorithm than then mrow element.

Additionally, it adds some rules for the UA stylesheet to properly render
these elements. This patch also makes the stylesheet align more closely
with the one defined in the spec
(<a href="https://w3c.github.io/mathml-core/#user-agent-stylesheet)">https://w3c.github.io/mathml-core/#user-agent-stylesheet)</a>

* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/mrow/legacy-mrow-like-elements-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/default-properties-on-semantics-and-maction-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/writing-mode/writing-mode-002-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/ignored-properties-001-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/html5-tree/dynamic-childlist-002-expected.txt:
* Source/WebCore/css/mathml.css:
(semantics &gt; :not(:first-child)):
(maction &gt; :not(:first-child)):
(merror):
(ms, mtext, mi, mn, mo, annotation, mtd):
(mfrac):
(mfrac &gt; *):
(mfrac &gt; :nth-child(2)):
(mroot &gt; *:last-child):
(mroot, msqrt):
(msub &gt; * + *, msup &gt; * + *, msubsup &gt; * + *, mmultiscripts &gt; * + *, munder &gt; * + *, mover &gt; * + *, munderover &gt; * + *):
(mfrac &gt; :nth-child(2),): Deleted.
* Source/WebCore/mathml/MathMLAnnotationElement.cpp:
(WebCore::MathMLAnnotationElement::createElementRenderer):
(WebCore::MathMLAnnotationElement::childShouldCreateRenderer const):
* Source/WebCore/mathml/MathMLSelectElement.cpp:
(WebCore::MathMLSelectElement::childShouldCreateRenderer const):
(WebCore::MathMLSelectElement::updateSelectedChild):
(WebCore::MathMLSelectElement::defaultEventHandler):
(WebCore::MathMLSelectElement::willRespondToMouseClickEventsWithEditability const):
</pre>

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/656f9bee6ab4c7ea65104d9ad8cbc94b00c04578. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50992

eerii committed:
Draft: Align UA Stylesheet with MathML Core for maction and semantics.
https://bugs.webkit.org/show_bug.cgi?id=262696

Reviewed by NOBODY (OOPS!).

Draft to run CI and check test results, probably needs cleanup later.
Align the maction and semantics parts of the MathML UA Stylesheet with MathML
Core (https://w3c.github.io/mathml-core/#user-agent-stylesheet).
Additionally, add headers to existing sections to more closely match the spec.

No new tests (OOPS!).

* Source/WebCore/css/mathml.css:
(semantics > :not(:first-child)):
(maction > :not(:first-child)):
(merror):
(ms, mtext, mi, mn, mo, annotation, mtd):
(mfrac):
(mfrac > *):
(mfrac > :nth-child(2)):
(mroot > *:last-child):
(mroot, msqrt):
(msub > * + *, msup > * + *, msubsup > * + *, mmultiscripts > * + *, munder > * + *, mover > * + *, munderover > * + *):
(mfrac > :nth-child(2),): Deleted.
* Source/WebCore/mathml/mathtags.in:

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4ba2e8677b04b26ad421a82924ca595518e49d4e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50992
